### ISSUE TITLE
cockpit window effect

### ISSUE THREAD

**Issue by gilbertohasnofb** [c172p-team/c172p#650]
Did we implement (or plan to do so) this cockpit window effect seen at around 9:13 in the video below?

https://www.youtube.com/watch?v=Dl756pvpiIA

I think it looks quite nice and it would give a more realistic feeling when flying "into the sun".

**Comment by gilbertohasnofb** [comment]
@wlbragg do you know anything about this?

**Comment by wlbragg** [comment]
Yes, it is something I want to do but was having trouble implementing it. I will try again and see if I can get it to work. One thing I could use though is the png mask that causes the glare to be more pronounced.

**Comment by gilbertohasnofb** [comment]
Where is our current png mask? I can try to improve it, but it's difficult for me to do so because I don't understand very well how the effects work :cry:

**Comment by onox** [comment]
Is it the [Mie forward scattering](http://wiki.flightgear.org/ALS_technical_notes#Mie_scattering) effect you mean?

We have `<texture n="3">` in `Models/Effects/interior/c172p-interior-glass-reflection-front.eff`, but don't we also need `<texture n="2">` if we're going to use [functional mask](http://wiki.flightgear.org/ALS_technical_notes#Functional_masks)?

**Comment by onox** [comment]
Btw, are our windows UV-mapped?

**Comment by gilbertohasnofb** [comment]
> Is it the Mie forward scattering effect you mean?

I really don't know the name of the effect, I just know that it's the pretty blurred light around the sun when viewed at direct angles in that video.

> Btw, are our windows UV-mapped?

I don't know, sorry.

**Comment by wlbragg** [comment]
>Is it the Mie forward scattering effect you mean?

Yes, I think so.

>Btw, are our windows UV-mapped?

Not positive, but I think so.

**Comment by wlbragg** [comment]
What aircraft was that example? I can look at it and get a better idea of the code and the mask makeup they used.

**Comment by gilbertohasnofb** [comment]
I think it's Richard's F15.

**Comment by wlbragg** [comment]
>don't we also need <texture n="2">

Yes.

What I was working on was a mask for defrost and defogging air vents. I remember, I never got anything to appear but I also didn't work on it that long.

**Comment by wlbragg** [comment]
I'm really wanting to implement this!
![glare](https://user-images.githubusercontent.com/11069680/36339103-95e38436-1383-11e8-99d5-eb97c9ccaa2f.png)

**Comment by gilbertohasnofb** [comment]
👍👍👍

**Comment by gilbertohasnofb** [comment]
@wlbragg would you like to consider to work on this next?

**Comment by wlbragg** [comment]
I still want to try to add this before closing the fog/frost PR. I really notice how "clean" the glass looks at all times except fog and frost and I think it should scuff up at least a little when looking into the glare.

**Comment by gilbertohasnofb** [comment]
@wlbragg if I would come up with a very light texture with smudges and small marks here and there, could we apply it to the glass with a very high alpha? Here is a photo I took the other day of a very dirty window:

![image](https://user-images.githubusercontent.com/5700795/49121930-44f3f180-f2aa-11e8-8ba2-5f05da0bae37.png)

So I was thinking something like this but less pronounced. It just would be cool to have this effect where the glass looks almost perfectly clean unless the sun hits it in the correct angle.

**Comment by wlbragg** [comment]
Yes, I agree. You might get a good idea of what we need by looking at the dirt mask used on the CRT's in the space shuttle.

**Comment by gilbertohasnofb** [comment]
Sounds good, I will take a look at those this weekend

**Comment by tigert** [comment]
This is caused by a worn out plastic on the windshield, which is pretty common on older trainer aircraft. It happens because repeated window cleaning (possibly with more or less dirty rugs) causes abrasion and scratches on the surface, making it a bit matt. It's indeed happening when you look directly at the sun and it is super annoying as it can spread throughout pretty much the whole windshield if it is in poor condition. At some point one just has to change the windshield to fix the problem.

**Comment by wlbragg** [comment]
This never was totally addressed in #1208
I still want to investigate the possibility of adding it, which I think we can.
